internal/describer/object.go

```diff
@@ -49,82 +51,80 @@ type Object struct {
 	objectType            func() interface{}
 	objectStoreKey        store.Key
 	disableResourceViewer bool
-	tabFuncDescriptors    []tabFuncDescriptor
+	tabFuncDescriptors    []Tab
 	iconName              string
 	iconSource            string
-	rootPath              ResourceLink
+	tabsGenerator         TabsGenerator
 }
 
 // NewObject creates an instance of Object.
 func NewObject(c ObjectConfig) *Object {
-	o := &Object{
-		path:                  c.Path,
-		baseTitle:             c.BaseTitle,
-		base:                  newBaseDescriber(),
-		objectStoreKey:        c.StoreKey,
-		objectType:            c.ObjectType,
-		disableResourceViewer: c.DisableResourceViewer,
-		iconName:              c.IconName,
-		iconSource:            c.IconSource,
-		rootPath:              c.RootPath,
+	tg := c.TabsGenerator
+	if tg == nil {
+		tg = NewObjectTabsGenerator()
 	}
 
-	o.tabFuncDescriptors = []tabFuncDescriptor{
-		{name: "Summary", tabFunc: o.addSummaryTab},
-		{name: "Metadata", tabFunc: o.addMetadataTab},
-		{name: "Resource Viewer", tabFunc: o.addResourceViewerTab},
-		{name: "YAML", tabFunc: o.addYAMLViewerTab},
-		{name: "Logs", tabFunc: o.addLogsTab},
-		{name: "Terminal", tabFunc: o.addTerminalTab},
+	td := c.TabDescriptors
+	if td == nil {
+		td = DefaultObjectTabs()
 	}
 
-	return o
-}
-
-type tabFunc func(ctx context.Context, object runtime.Object, cr *component.ContentResponse, options Options) error
+	o := &Object{
+		path:               c.Path,
+		baseTitle:          c.BaseTitle,
+		base:               newBaseDescriber(),
+		objectStoreKey:     c.StoreKey,
+		objectType:         c.ObjectType,
+		iconName:           c.IconName,
+		iconSource:         c.IconSource,
+		tabsGenerator:      tg,
+		tabFuncDescriptors: td,
+	}
+
+	o.tabFuncDescriptors = []Tab{
+		{Name: "Summary", Factory: SummaryTab},
+		{Name: "Metadata", Factory: MetadataTab},
+		{Name: "Resource Viewer", Factory: ResourceViewerTab(false)}, // TODO: Fix me
```

Comment:
can we create an issue for this fix me with what needs to be fixed, can leave the TODO but have it reference a GH issue.